app/views/Dashboard/index.tsx

```diff
@@ -310,6 +404,48 @@ function Dashboard() {
         indicatorList,
     ]);
 
+    useEffect(() => {
+        if (!pendingExport) {
+            if (exportData?.length > 0) {
+                if (exportData.length === exportTotal) {
+                    /*
+                    const dataString = stringify(exportData, {
+                        columns: headers,
+                    });
+                    */
+                    const blob = new Blob(
+                        [exportFullString],
+                        { type: 'text/csv' },
+                    );
+                    saveAs(blob, 'Data Export');
+                } else {
+                    // eslint-disable-next-line no-console
+                    console.error('CSV num rows mismatch', `expected: ${exportTotal}`, `got: ${exportData.length}`);
+                }
+            }
+        }
```

Comment:
Too must nested ifs.
Let's do this on onCompleted instead of an useEffect.

Reply:
done!

Reply:
much*